packages/amplify-graphql-transformer-core/src/migration/migrators/auth/ownerAuth.ts

```diff
@@ -0,0 +1,64 @@
+import { getAuthRules } from '.'
+import { createAuthRule } from '../generators'
+
+function getPrivateAuthRule(rules: any, provider: any) {
+  return rules.find((rule: any) => {
+    const foundStrategy = rule.fields.find((f: any) => f.name.value === "allow").value.value
+    const foundProvider = rule.fields.find((f: any) => f.name.value === "provider")?.value?.value
+
+    if (foundStrategy !== 'private') {
+      return false
+    }
+
+    if (provider === "userPools" && (foundProvider === undefined || foundProvider === "userPools")) {
+      return true
+    } else if (provider === "iam" && foundProvider === "iam") {
+      return true
+    }
+    return false
+  })
+}
+
+function getOwnerAuthRules(rules: any) {
+  return rules.filter((rule: any) => rule.fields.find((f: any) => f.name.value === 'allow').value.value === 'owner')
+}
+
+export function migrateOwnerAuth(node: any, defaultAuthMode: any) {
+  const authRules = getAuthRules(node)
+
+  // check if owner-based auth exist and if operations allow everything.
+  const deniedOperations = new Set()
+  const ownerRules = getOwnerAuthRules(authRules)
+  if (ownerRules.length !== 0) {
+    ownerRules.forEach((rule: any) => {
+      const operationsIndex = rule.fields.findIndex((f: any) => f.name.value === 'operations')
+
+      if (operationsIndex === -1) {
+        deniedOperations.add('update')
+        deniedOperations.add('read')
+        deniedOperations.add('delete')
+      } else {
+        // remember denied operations
+        rule.fields[operationsIndex].value.values.map((op: any) => op.value).forEach((op: any) => deniedOperations.add(op))
```

Comment:
Can the `map()` and `forEach()` be combined into a single `forEach()` where you add `op.value` to the set?